packages/flutter/lib/src/widgets/modal_button_sheet.dart

```diff
@@ -0,0 +1,652 @@
+import 'dart:async';
+import 'dart:ui';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+
+const Curve _decelerateEasing = Cubic(0.0, 0.0, 0.2, 1.0);
+const Curve _modalBottomSheetCurve = _decelerateEasing;
+const Duration _bottomSheetDuration = Duration(milliseconds: 400);
+const double _minFlingVelocity = 500.0;
+const double _closeProgressThreshold = 0.6;
+const double _willPopThreshold = 0.8;
+
+typedef WidgetWithChildBuilder = Widget Function(
+    BuildContext context, Animation<double> animation, Widget child);
+
+/// A custom bottom sheet.
+///
+/// The [ModalBottomSheet] widget itself is rarely used directly. Instead, prefer to
```

Comment:
Same as mentioned on top. I would perhaps not use the "bottom sheet" terminology since they already have a specific meaning.